cocotb/_py_compat.py

```diff
@@ -32,7 +32,7 @@
 
 
 # backport of Python 3.7's contextlib.nullcontext
-class nullcontext(object):
+class nullcontext():
```

Comment:
```suggestion
class nullcontext:
```
Is the style I'm more used to. Not a big deal

Reply:
I'm indifferent, but there are existing classes that use empty parens, so that's what I went with.

Reply:
Python docs seem to strongly prefer `class C:` over `class C():`, so I'd appreciate if you could change this.

Reply:
Everywhere, or just on the classes I've touched?

Reply:
Up to you

Reply:
Scope crept.